modules/s3/src/main/clojure/xtdb/s3/file_list.clj

```diff
@@ -86,6 +86,9 @@
       (:Records)
       (first)))
 
+(defn url-decode-filename [filename]
+  (string/replace filename "%24" "$"))
+
```

Comment:
Can me make sure this is the only problem? I think there could be other funky table names etc..

Reply:
Used `URLDecoder` to cover our bases here.